### DISCUSSION TITLE
SSL Connect error

### DISCUSSION THREAD

**Discussion by Bab95** [curl/curl#13024] [Q&A] [answered]
I am getting SSL Connect error. following are the verbose logs

```
:Failed sending curl request with error:SSL connect error
== Info:   Trying <ip>:443...
== Info: Connected to <url> (<ip>) port 443
== Info: ALPN: curl offers http/1.1
=> Send SSL data: 
```

I am confused here when curl says connected to <url> does that mean curl was able to reach to the server? and then some failure happened while receiving data from server side? Or it wasn't ?

What I am trying to make understanding around is that if only inbound traffic is being blocked and outbound requests are allowed or both are blocked.

**Comment by icing** [comment, marked as answer]
When `curl` says `Connected to address port`, it means it has established a TCP connection to the server. Thereafter, for `https:` and other urls it will start the "SSL handshake" where both sides exchange information and the server has to prove that it can be trusted.

**Reply by Bab95** [reply]
Means client send hello message but didn't hear it back? means inbound traffic is blocked? Is this assumption correct?

**Reply by icing** [reply]
I'd need to see the complete log to answer that. Also, if you are interested in seeing SSL/TCP traces and your curl is recent, you may invoked curl with `curl -v --trace-config ids,time,tcp,ssl`.

**Reply by Bab95** [reply]
I am using LibCurl actually where i have set VERBOSE Flag to 1. is this --trace-config adding more detailed traces?

**Reply by jay** [reply]
> is this --trace-config adding more detailed traces?

Yeah. Recent versions of libcurl have [curl_global_trace](https://curl.se/libcurl/c/curl_global_trace.html) which does pretty much the same thing.
